test/static.js

```diff
@@ -304,3 +381,242 @@ test('/documentation/:file should be served from custom location with trailing s
+      path: './examples/example-static-specification.yaml'
+    },
+    exposeRoute: true
+  }
+
+  t.plan(8)
+  const fastify = Fastify()
+  fastify.register(fastifySwagger, config)
+
+  fastify.inject(
+    {
+      method: 'GET',
+      url: '/documentation/yaml'
+    },
+    (err, res) => {
+      t.error(err)
+      t.is(typeof res.payload, 'string')
+      t.is(res.headers['content-type'], 'application/x-yaml')
+      try {
+        yaml.safeLoad(res.payload)
+        t.pass('valid swagger yaml')
+      } catch (err) {
+        t.fail(err)
+      }
+    }
+  )
+
+  fastify.inject(
+    {
+      method: 'GET',
+      url: '/documentation/yaml'
+    },
+    (err, res) => {
+      t.error(err)
+      t.is(typeof res.payload, 'string')
+      t.is(res.headers['content-type'], 'application/x-yaml')
+      try {
+        yaml.safeLoad(res.payload)
+        t.pass('valid swagger yaml')
+      } catch (err) {
+        t.fail(err)
+      }
+    }
+  )
+})
+
+test('/documentation/json returns cache.swaggerObject on second request in dynamic mode', t => {
+  const config = {
+    specification: {
+      path: './examples/example-static-specification.json'
+    },
+    exposeRoute: true
+  }
+
+  t.plan(8)
+  const fastify = Fastify()
+  fastify.register(fastifySwagger, config)
+
+  fastify.inject(
+    {
+      method: 'GET',
+      url: '/documentation/json'
+    },
+    (err, res) => {
+      t.error(err)
+      t.is(typeof res.payload, 'string')
+      t.is(res.headers['content-type'], 'application/json; charset=utf-8')
+      try {
+        t.pass('valid swagger json')
+      } catch (err) {
+        t.fail(err)
+      }
+    }
+  )
+
+  fastify.inject(
+    {
+      method: 'GET',
+      url: '/documentation/json'
+    },
+    (err, res) => {
+      t.error(err)
+      t.is(typeof res.payload, 'string')
+      t.is(res.headers['content-type'], 'application/json; charset=utf-8')
+      try {
+        t.pass('valid swagger json')
+      } catch (err) {
+        t.fail(err)
+      }
+    }
+  )
+})
+
+test('prints to console when swagger routes are not exposed', t => {
+  const config = {
+    mode: 'static',
+    specification: {
+      path: './examples/example-static-specification.json'
+    },
+    exposeRoute: false
+  }
+
+  t.plan(4)
+  const fastify = Fastify()
+  fastify.register(fastifySwagger, config)
+
+  // check that yaml is there
+  fastify.inject(
+    {
+      method: 'GET',
+      url: '/documentation/json'
+    },
+    (err, res) => {
+      t.error(err)
+      t.is(typeof res.payload, 'string')
+      t.is(res.headers['content-type'], 'application/json; charset=utf-8')
+      try {
+        t.pass('routes are not exposed')
+      } catch (err) {
+        t.fail(err)
+      }
+    }
+  )
+})
+
+test('inserts default opts in fastifySwagger', t => {
+  t.plan(1)
+  const fastify = Fastify()
+  const next = () => console.log()
+
+  // eslint-disable-next-line
+  fastify.register(Function(fastifySwagger(fastify, null, next)))
```

Comment:
Why the `Function`?

Reply:
If we don't use a `Function` constructor, `avvio` throws an error saying that:
```
 FAIL  test/static.js
✖ plugin must be a function or a promise
node_modules/avvio/boot.js                                                                       
  192 | function assertPlugin (plugin) {                                                           
  193 |   if (!(plugin && (typeof plugin === 'function' || typeof plugin.then === 'function'))) {  
> 194 |     throw new Error('plugin must be a function or a promise')                              
      | ----------^                                                                                
  195 |   }                                                                                        
  196 | }                                                                                          
  197 | 
```

So, if we remove the `Function` for this test, it'll fail